storybookjs/storybook
# [WIP] Addon readme

## What I did

Created an addon based on the `notes` addon for adding a readme to all stories within a `storiesOf` via a decorator, including github-style formatting

## Why this is necessary

The notes addon is designed to allow users to add a note to a particular story, this is designed to allow for a readme that applies to a whole component (or group of components)

The ReadMe panel clearly delineates between general notes and a ReadMe file that teams can use to document things like implementation details, why a decision was made, etc.

Universal support - as this does not have any impact on rendering it should work across all of the app platforms (I need to confirm this)

## Example:
```jsx
import React from 'react';
import { withReadMe } from '@storybook/addon-readme';
import { storiesOf } from '@storybook/react';
import BaseButton from '../components/BaseButton';
import readMe from './notes/readme.md';

storiesOf('Addons|ReadMe', module)
  .addDecorator(withReadMe(readMe))
  .add('with text', () => <BaseButton label="Button with readme" />);
```

## Screenshot

<img width="1280" alt="screen shot 2018-01-29 at 8 26 26 pm" src="https://user-images.githubusercontent.com/4509692/35543366-ba243e4e-0532-11e8-8287-a34751d73564.png">


## How to test

Is this testable with jest or storyshots?

Does this need a new example in the kitchen sink apps?
yes

Does this need an update to the documentation?
yes

If your answer is yes to any of these, please make sure to include it in your PR.


## TODO:

- [x] Add an example to the storybook-official app
- [ ] Add a proper README
